modules/consul-cluster/main.tf

```diff
@@ -39,10 +39,15 @@ resource "aws_autoscaling_group" "autoscaling_group" {
       value               = "${var.cluster_tag_value}"
       propagate_at_launch = true
     },
-    "${var.tags}",
+    "${var.tags}"
   ]
 }
 
+resource "aws_autoscaling_attachment" "consul-asg-tg-attachment" {
+  autoscaling_group_name = "${aws_autoscaling_group.autoscaling_group.name}"
+  alb_target_group_arn   = "${var.target_group_arn}"
```

Comment:
The "alb_target_group_arn" argument is [optional](https://www.terraform.io/docs/providers/aws/r/autoscaling_attachment.html). I tried it and the terraform created an attachment without the argument (see output below). Note the attachment is still referenced by the ASG, it just doesn't contain any value. Also verified the field *Target Groups* was empty in the AWS Console ASG UI. Note this field is a list, so since we don't add anything to it, I expect it will remain an empty list.
`  + module.prod_consul_1.module.consul_servers.aws_autoscaling_attachment.consul-asg-tg-attachment
      id:                                        <computed>
      autoscaling_group_name:                    "${aws_autoscaling_group.autoscaling_group.name}"
---> NOTE "alb_target_group_arn" is missing from here, which is where it shows up if specified

  + module.prod_consul_1.module.consul_servers.aws_autoscaling_group.autoscaling_group
      id:                                        <computed>
      arn:                                       <computed>
      default_cooldown:                          <computed>
      desired_capacity:                          "3"
      ...
      target_group_arns.#:                       <computed>
      ...
`